Samples.Objc/Samples.ObjC/ViewControllers/CheckoutViewController/CheckoutViewController.m

```diff
@@ -0,0 +1,376 @@
+}
+
+- (NSString *)getItemTotal {
+  double total = 0;
+  for (PPCPurchaseUnitItem* item in self.items) {
+    double price = [item.unitAmount.value doubleValue];
+    double quantity = [item.quantity doubleValue];
+    total += price * quantity;
+  }
+  return [[NSNumber numberWithDouble:total] stringValue];
+}
+
+- (NSString *)getTaxTotal {
+  double total = 0;
+  for (PPCPurchaseUnitItem* item in self.items) {
+    double tax = [item.tax.value doubleValue];
+    double quantity = [item.quantity doubleValue];
+    total += tax * quantity;
+  }
+  return [[NSNumber numberWithDouble:total] stringValue];
+}
+
+// MARK: - UITableViewDelegate and UITableViewDataSource
+- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
+  return 2;
+}
+
+- (NSString *)tableView:(UITableView *)tableView titleForHeaderInSection:(NSInteger)section {
+  switch (section) {
+    case 0:
+      return @"Items";
+    case 1:
+      return @"Cart total";
+    default:
+      return @"";
+  }
+}
+
+- (nonnull UITableViewCell *)tableView:(nonnull UITableView *)tableView cellForRowAtIndexPath:(nonnull NSIndexPath *)indexPath {
+  if (indexPath.section == 0) {
+    if (indexPath.row >= self.items.count) {
+      AddItemCell *cell = [[AddItemCell alloc] init];
+      return cell;
+    } else {
+      ItemCell *cell = [[ItemCell alloc] initWithItem:self.items[indexPath.row] andQuantity:self.items[indexPath.row].quantity];
+      return cell;
+    }
+  } else {
+    TotalCell *cell = [[TotalCell alloc] initWithTotal:[self getTotal] subtotal:[self getItemTotal] tax:[self getTaxTotal]];
+    return cell;
+  }
+}
+
+- (CGFloat)tableView:(UITableView *)tableView heightForRowAtIndexPath:(NSIndexPath *)indexPath {
+  if (indexPath.section == 0) {
+    if (indexPath.row >= self.items.count) {
+      return 48;
+    }
+    return UITableViewAutomaticDimension;
+  } else {
+    return UITableViewAutomaticDimension;
+  }
+}
+
+- (NSInteger)tableView:(nonnull UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
+  if (section == 0) {
+    return self.items.count + 1;
+  } else {
+    return 1;
+  }
+}
+
+- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
+  if (indexPath.section == 0) {
+    if (indexPath.row >= self.items.count) {
+      AddItemViewController *vc = [[AddItemViewController alloc] init];
+      vc.delegate = self;
+      vc.index = indexPath.row;
+      [self presentViewController:vc animated:true completion:^{}];
+    } else {
+      AddItemViewController *vc = [[AddItemViewController alloc] initWithItem:self.items[indexPath.row]];
+      vc.delegate = self;
+      vc.index = indexPath.row;
+      [self presentViewController:vc animated:true completion:^{}];
+    }
+  }
+}
+
+- (BOOL)tableView:(UITableView *)tableView canEditRowAtIndexPath:(NSIndexPath *)indexPath {
+  if (indexPath.section == 0 && indexPath.row < self.items.count) {
+    return true;
+  }
+  return false;
```

Comment:
can be shorted to 
`return (indexPath.section == 0 && indexPath.row < self.items.count)`